### PR TITLE
enable chained config repo

### PR DESCRIPTION
**Pull request by lukas-c-wilhelm**
Previously, the repositories worked only when created as Spring beans due to the injection of the `ConfigurationParser`. This was at odds with their constructors being accessible. In particular, this broke usage of the ChainedConfigurationRepository if the repositories passed to it were not created as beans themselves.

This is solved in this commit by adding a `RepositoryFactory` bean through which all repositories are created.

Also, the class `ConfigTypeProvider` is removed as it was an unnecessary layer just wrapping `BaiganConfigClasses`.

### CODE REVIEW

File: src/main/java/org/zalando/baigan/service/RepositoryFactory.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    public EtcdConfigurationRepository etcdConfigurationRepository(String etcdUrl) {
+        return new EtcdConfigurationRepository(etcdUrl, configurationParser);
+    }
+
+    public ChainedConfigurationRepository chainedConfigurationRepository(List<ConfigurationRepository> configurationRepositories) {
+        if (configurationRepositories.size() == 2) {
+            return new ChainedConfigurationRepository(configurationRepositories.get(0), configurationRepositories.get(1));
+        } else if (configurationRepositories.size() > 2) {
```
Comment (lukasniemeier-zalando) [inline src/main/java/org/zalando/baigan/service/RepositoryFactory.java:34]: Shall we gift `ChainedConfigurationRepository` a new constructor that handles our arbitrary sized list a little bit more convenient?

File: src/main/java/org/zalando/baigan/service/RepositoryFactory.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    public S3ConfigurationRepositoryBuilder s3ConfigurationRepository() {
+        return new S3ConfigurationRepositoryBuilder(configurationParser);
+    }
+
+    public FileSystemConfigurationRepositoryBuilder fileSystemConfigurationRepository() {
+        return new FileSystemConfigurationRepositoryBuilder(configurationParser);
+    }
+
+    public EtcdConfigurationRepository etcdConfigurationRepository(String etcdUrl) {
+        return new EtcdConfigurationRepository(etcdUrl, configurationParser);
```
Comment (lukasniemeier-zalando) [inline src/main/java/org/zalando/baigan/service/RepositoryFactory.java:28]: Is there a reason why this one is an actual repository (while the other we have migrated to _builders_)? Worth to also provide a builder - for the sake of symmetry - here or does it not make sense before migrating the Etcd repository to a modern version?

**Review by lukasniemeier-zalando — APPROVED** [review]

File: src/main/java/org/zalando/baigan/service/RepositoryFactory.java (hunk 2 above)
Comment (lukas-c-wilhelm) [reply src/main/java/org/zalando/baigan/service/RepositoryFactory.java:28]: Done in https://github.com/zalando-stups/baigan-config/pull/84

File: src/main/java/org/zalando/baigan/service/RepositoryFactory.java (hunk 1 above)
Comment (lukas-c-wilhelm) [reply src/main/java/org/zalando/baigan/service/RepositoryFactory.java:34]: Done in https://github.com/zalando-stups/baigan-config/pull/84

**Comment by lukas-c-wilhelm** [comment]
Closing this PR. I created PR https://github.com/zalando-stups/baigan-config/pull/84 instead that will be merged before the introduction of structured configs. The remaining changes of the present PR will be added to https://github.com/zalando-stups/baigan-config/pull/77 directly instead.

**Pull request closed without merging**